### PR TITLE
Add Map Extension Method to Support Mapping from Result<T> to Result

### PR DESCRIPTION
**Pull request by jmrtnz94**
See https://github.com/ardalis/Result/issues/212

### CODE REVIEW

File: src/Ardalis.Result/ResultExtensions.cs [hunk 1]
```diff
@@ -43,5 +43,39 @@ public static Result<TDestination> Map<TSource, TDestination>(this Result<TSourc
...
+                    : Result.Unauthorized();
+                case ResultStatus.Forbidden: return result.Errors.Any()
+                    ? Result.Forbidden(result.Errors.ToArray())
+                    : Result.Forbidden();
+                case ResultStatus.Invalid: return Result.Invalid(result.ValidationErrors);
+                case ResultStatus.Error: return Result.Error(new ErrorList(result.Errors.ToArray(), result.CorrelationId));
+                case ResultStatus.Conflict: return result.Errors.Any()
+                    ? Result.Conflict(result.Errors.ToArray())
+                    : Result.Conflict();
+                case ResultStatus.CriticalError: return Result.CriticalError(result.Errors.ToArray());
```
Comment (christophercalm) [inline src/Ardalis.Result/ResultExtensions.cs:73]: Does this need the correlation id as well?

Comment (jmrtnz94) [reply src/Ardalis.Result/ResultExtensions.cs:73]: CriticalError accepts a string[], so we cannot pass in the correlation id. See https://github.com/ardalis/Result/blob/main/src/Ardalis.Result/Result.cs#L223.

While Error accepts an ErrorList object that contains correlation id. See https://github.com/ardalis/Result/blob/main/src/Ardalis.Result/Result.cs#L123-L127

Comment (KyleMcMaster) [reply src/Ardalis.Result/ResultExtensions.cs:73]: I believe we're in a bit of a transition phase to using the `ErrorList` class in more places but ideally we'd use the constructors that accept a `CorrelationId` wherever possible.

Comment (KyleMcMaster) [reply src/Ardalis.Result/ResultExtensions.cs:364]: I have verified the CorrelationId property is propagated through the constructors where it is available. We can open another issue to add CorrelationId to the remaining constructors and then update these helper methods as needed.

**Review by KyleMcMaster — COMMENTED** [review]
try fix build

**Review by KyleMcMaster — APPROVED** [review]

**Pull request merged**